### PR TITLE
Revise models docs to indicate attributes are copied for validation

### PR DESCRIPTION
**Pull request by tpdorsey**
Closes https://github.com/pydantic/pydantic/issues/1068

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by tpdorsey** [comment]
please review

**Review by Kludex — DISMISSED** [review]

File: docs/usage/dataclasses.md [hunk 1]
```diff
@@ -377,3 +377,34 @@ print(pydantic_core.to_json(user, indent=4).decode())
...
+
+In this example, note that the ID of the list changes after the class is constructed because it has been copied for validation.
+
+```py
+from typing import List
+
+from pydantic import BaseModel
+
+
+class C1:
```
Comment (Kludex) [inline docs/usage/dataclasses.md:392]: Was this supposed to be a dataclass? There's no reference to dataclasses in this section. 🤔

File: docs/usage/dataclasses.md [hunk 2]
```diff
@@ -377,3 +377,34 @@ print(pydantic_core.to_json(user, indent=4).decode())
 }
 """
 ```
+## Attribute copies
+
+As described earlier, when constructing classes with data attributes, Pydantic copies the the attribute in order to efficiently iterate over its elements for validation.
```
Comment (Kludex) [inline docs/usage/dataclasses.md:382]:
```suggestion
As described earlier, when constructing classes with data attributes, Pydantic copies the attributes in order to efficiently iterate over its elements for validation.
```

File: docs/usage/dataclasses.md (hunk 1 above)
Comment (tpdorsey) [reply docs/usage/dataclasses.md:392]: You are totally correct. Moving this to models

**Comment by tpdorsey** [comment]
please review

**Comment by tpdorsey** [comment]
please review

**Review by hramezani — APPROVED** [review]

**Pull request merged**